### DISCUSSION TITLE
Implementation of BO

### DISCUSSION THREAD

**Discussion by SamirLamin** [meta-pytorch/botorch#1649] [Q&A]
I have an optimization problem with 3 objectives and 2 variables decisions and the functios are a black box. I need to implement BO using Botorch to solve my problem so I need to create a class of my problem in multi_objective. 
My question is how I can create my problem like other problems and respect the same way to implement BO.
?
My second question the evaluate_true is the function that evaluates a solution to find the best solution?

**Comment by eytan** [comment]
Have you tried checking out http://ax.dev —it’s our easy to use interface
for practitioners who want to do black box optimization. BoTorch is a lower
level framework targeted to people who are developing new Bayesian
optimization methods. Ax should be easier to get started.

**Reply by SamirLamin** [reply]
Ok, thank you so much  I will check it. but  where and how I can add my problem  and Implement the BO

**Reply by saitcakmak** [reply]
Hi @SamirLamin. Ax Service API is probably the easiest way for you to get started. Section 2 of the tutorial shows how tosetup the optimizer with your search space and objectives, and section 3 is where you'd hook up the objective evaluations (could be any arbitrary python code that evaluates your problem): https://ax.dev/tutorials/gpei_hartmann_service.html
